### ISSUE TITLE
[v3.8.1] AisHierarchicalMenu CreateURL generates different URL when selected vs deselected - can hurt SEO crawlers.

### ISSUE THREAD

**Issue by jonathan-bird** [algolia/vue-instantsearch#1132]
**Bug 🐞**

### What is the current behavior?
When a category is not selected, the URL will be the current category URL (expected), and when you select the category, the URL for that category then changes to the URL of the parent, to de-select that category and go back to the parent category. This may be nice behaviour for the user (possibly - may be nice to have as a boolean flag though), however it would cause issues with Google/SEO crawlers as it creates inconsistencies with what a page URL should actually be.

If you think about how an SEO crawler works, they go through every URL and match up words to URLs and work out backlinks etc. This can be confusing for a crawler if it's different on each page.

### Make a sandbox with the current behavior

https://codesandbox.io/s/vue-instantsearch-double-query-request-zp7v6?file=/src/App.vue

### What is the expected behavior?

To either keep the same URL, or have an option to toggle between the behaviour.

### Does this happen only in specific situations?

No

### What is the proposed solution?

As above.

### What is the version you are using?

Vue 2.x, InstantSearch v3.8.1

Please let me know if it's different behaviour on IS v4.

**Comment by jonathan-bird** [comment]
Yeah that's correct. I was having a think about this earlier and I think only Hierarchical Menu's at this stage. As Refinement lists are checkboxes, as you say, it makes sense to work as it currently does where it goes back to the parent when deselecting. But not when it's a radio btn/html link.

**Comment by jonathan-bird** [comment]
Please let me know if I can help in any way with this @Haroenv :)

**Comment by jonathan-bird** [comment]
Do you know if this planned on being looked at @Haroenv?

**Comment by Haroenv** [comment]
We've not found a consensus on what the correct behaviour is, sorry. I'd recommend using a custom widget where you call `createURL(item.isRefined ? '' : item.value)` (pseudo code) to avoid the default behaviour @jonathan-bird. Sorry that I forgot to get back on this issue
